src/Visitor/MutationsCollectorVisitor.php

```diff
@@ -72,10 +72,17 @@ public function leaveNode(Node $node)
 
             $isOnFunctionSignature = $node->getAttribute(ReflectionVisitor::IS_ON_FUNCTION_SIGNATURE, false);
 
-            if (!$isOnFunctionSignature) {
-                if (!$node->getAttribute(ReflectionVisitor::IS_INSIDE_FUNCTION_KEY)) {
-                    continue;
-                }
+            if (!$isOnFunctionSignature
+                && !$node->getAttribute(ReflectionVisitor::IS_INSIDE_FUNCTION_KEY)
+            ) {
+                continue;
+            }
+
+            if ($isOnFunctionSignature
```

Comment:
hm..

[we have a test](https://github.com/infection/infection/blob/master/tests/Fixtures/Autoloaded/PublicVisibility/pv-same-method-abstract.php) that proofs it should not be mutated on the mutator level:

https://github.com/infection/infection/blob/638e06296271fdb10193159cec0a8139af1ce9d8/tests/Mutator/FunctionSignature/PublicVisibilityTest.php#L178-L180

Probably, something is broken in another place